src/widget/widgetengine.js

```diff
@@ -34,6 +35,22 @@ export default class WidgetEngine extends Plugin {
 			}
 
 			const viewSelection = conversionApi.viewSelection;
+
+			// Add CSS class if selection is placed inside nested editable that belongs to widget.
+			const editableElement = viewSelection.editableElement;
+
+			if ( editableElement && !( editableElement instanceof RootEditable ) ) {
+				const widget = editableElement.findAncestor( element => isWidget( element ) );
+
+				if ( widget ) {
+					widget.addClass( WIDGET_SELECTED_CLASS_NAME );
```

Comment:
This is complicated :P.
Instead of using stuff like:
```
const STICKS_TO_PREVIOUS = 0;
const STICKS_TO_NEXT = 1;
...
if ( this.stickiness == STICKS_TO_PREVIOUS ) { ... }
```
we decided to just use `sticksToPrevious`. But this works only for closed list of constants and was done to get rid of imports.

I don't know whether we decided anything about package-scope/configuration constants like this. If I wrote that code, I'd go with `widgetSelectedClassName` but I can't say this is wrong either.